docs/guides/provider_guide.md

```diff
@@ -0,0 +1,5 @@
+# Guide: <guide name> Delphix Provider Guide
```

Comment:
Do we have two headers for this section? Is that needed?

Does the first one appear in the Table of Contents on the left hand side? The markdown tester doesn't show it.